File: programmers/etc/level2/유사_칸토어_비트열.py
def solution(n, l, r):
    answer = 0

    def check(idx):
        if idx%5 == 2:
            return 0
        elif idx < 5:
            return 1
        else:
            return check(idx//5)

    for i in range(l-1, r):
        answer += check(i)

    return answer


n, l, r = 2, 4, 17
print(solution(n, l, r)) # 8

File: Java/programmers/etc/level2/유사칸토어비트열.java
package CodeTest.Java.programmers.etc.level2;

public class 유사칸토어비트열 {
    public int solution(int n, long l, long r) {
        int answer = 0;
        
        for (long i=l-1; i<r; i++) {
            answer += check(i);
        }
        
        return answer;
    }
    
    private int check(long idx) {
        if (idx%5 == 2) return 0;
        else if (idx < 5) return 1;
        else return check(idx/5);
    }
}
